### PR TITLE
Necessary changes for HWRF physics (Ferrier-Aligo MP in dycore)

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

See https://github.com/ufs-community/ufs-weather-model/pull/223.

This PR contains two changes, one for avoiding crashes when the dycore debug option is turned on (in `model/fv_dynamics.F90`), and one for running microphysics with fewer tracers than in the input data set (in `tools/external_ic.F90`). This is required for Ferrier-Aligo microphysics.

## Testing

See https://github.com/ufs-community/ufs-weather-model/pull/223.

## Dependencies

https://github.com/ufs-community/ufs-weather-model/pull/223
https://github.com/NOAA-EMC/fv3atm/pull/188
https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/43
https://github.com/NCAR/ccpp-physics/pull/509

### CODE REVIEW

File: tools/external_ic.F90 [hunk 1]
```diff
@@ -570,8 +570,10 @@ subroutine get_nggps_ic (Atm, fv_domain, dt_atmos )
 
 !--- read in the number of tracers in the NCEP NGGPS ICs
       call read_data ('INPUT/'//trim(fn_gfs_ctl), 'ntrac', ntrac, no_domain=.TRUE.)
-      if (ntrac > ntracers) call mpp_error(FATAL,'==> External_ic::get_nggps_ic: more NGGPS tracers &
-                                 &than defined in field_table '//trim(fn_gfs_ctl)//' for NGGPS IC')
+      ! DH* 20200922 - this breaks Ferrier-Aligo MP runs
```
Comment (junwang-noaa) [inline tools/external_ic.F90:573]: How about other MP schemes? Can we comment out this error check line only for Ferrier_Aligo MP?

Comment (climbfuji) [reply tools/external_ic.F90:573]: There is no possibility to check for the microphysics scheme, the only check we could do is if `nwat` is 4 or not (but then it also applies to GFDL's Kessler scheme. This is the change for which I asked the GFDL dycore help desk last week (no reply yet).

Comment (junwang-noaa) [reply tools/external_ic.F90:573]: So, does this mean that in this case the ICs contain more tracers than Ferrier_Aligo MP requires?

Comment (climbfuji) [reply tools/external_ic.F90:573]: That's correct, the ICs are "NGGPS ICs" with the GFDL MP tracers, but FA only needs a subset of them.

**Comment by bensonr** [comment]
What are the science impacts of taking ICs for a different number of
hydrometeors / MP scheme and utilizing them in a reduced set?  I assume the
difference is in the number of water species for the F-A MP scheme.  Has
there been logic included to convert the missing water species to other
precipitation types for the reduced set in F-A?

**Comment by climbfuji** [comment]
> What are the science impacts of taking ICs for a different number of hydrometeors / MP scheme and utilizing them in a reduced set? I assume the difference is in the number of water species for the F-A MP scheme. Has there been logic included to convert the missing water species to other precipitation types for the reduced set in F-A?

These are good points. The HAFS group was experimenting with code that initializes the tracers based on what is available in the input files, but it hasn't been included in this PR. For the moment, we cannot conduct any physics testing with FA MP because the model crashes with a memory corruption after the first pass through the dycore. This problem occurred first when the GFDL_atmos_cubed_sphere repository was updated earlier this year (alongside with an FMS update, which was reverted as far as I remember). I did mention this a few times in the past, but we haven't gotten to the bottom of the problem yet.

Ideally, one would be running `chgres` to create the initial conditions for the MP scheme of choice, which would take care of adding the correct tracers (and doing any conversion of hydrometeors if needed). It is my understanding that the UFS_UTILS developers are working towards this. But independent of that we should be able to run the model with whatever MP scheme based on whatever hydrometeors are in the input data. Whether this is scientifically sound or not is a different question, but at least it allows one to conduct experiments and debug problems as the one above.

I had sent a question to the GFDL help desk last week whether removing the guard could be the reason for the stack corruption, but I haven't heard back yet. If the only reason the guard is there is to prevent physics inconsistencies, then I would be in favor of removing it or converting it into a warning. If, however, it has wider implications then we need to keep it and find another way to do our debugging and development with FA MP.

**Comment by climbfuji** [comment]
@bensonr did my reply to your comment make sense? Do you have further questions or concerns? We would like to merge the HWRF physics into the repositories today, if possible. Thanks!

**Comment by bensonr** [comment]
@climbfuji - I've checked with the extended team at GFDL and removal of the restriction is acceptable.

**Review by bensonr — APPROVED** [review]

**Comment by climbfuji** [comment]
> @climbfuji - I've checked with the extended team at GFDL and removal of the restriction is acceptable.

Thanks, @bensonr - appreciate your efforts!

**Review by junwang-noaa — APPROVED** [review]

**Review by XiaqiongZhou-NOAA — APPROVED** [review]

**Pull request merged**